docs/guide/initializing-dapps.md

```diff
@@ -1,20 +1,20 @@
 # Initializing Dapps
 
-Once you have your basic dev environment [set up](./Getting-Started), you are ready to start interacting with some smart contracts. There are some basic things you'll need regardless of what convenience library you're using, when communicating with a smart contract:
+Once you have your basic dev environment [set up](./Getting-Started), you are ready to start interacting with some smart contracts. There are some basic things you'll need, regardless of what convenience library you're using, when communicating with a smart contract:
 
 ## The Contract Network
 
-If you aren't connected to the right network, you aren't going to have any luck sending transactions to your contract, so make sure you have this right!
+If you aren't connected to the right network, you aren't going to have any luck sending transactions to your contract, so make sure you have this right! Many dapp developers choose to deploy their contract to a testnet first, in order to avoid potentially disastrous fees if something goes wrong during development and testing on mainnet.
 
-Many clever dapps will recognize the user's current network, and actually adapt to it! For example, if you detect a test network, you could deliberately connect to a test-network version of your smart contract, which makes it easy for users to "try out" your system without using real money!
+No matter which network you deploy your final dapp on, your user will need to be able to access it. As of v3.3.0, MetaMask makes available [wallet_switchEthereumChain](https://docs.metamask.io/guide/rpc-api.html#other-rpc-methods), allowing you to prompt the user to add a chain that you suggest, and switch to it using a confirmation dialogue.
```

Comment:
`wallet_switchEthereumChain` was added in v9.6.0 of the extension, and in v3.3.0 of mobile. Rather than explaining that, maybe we can just omit the release version?

Also, `wallet_switchEthereumChain` does not allow the user to add a chain that the dapp suggests. We have a separate method (`wallet_addEthereumChain`) for that.